src/text.rs

```diff
@@ -281,9 +284,15 @@ impl<'a> Text<'a> {
     {
         let lines: Vec<_> = match content.into() {
             Cow::Borrowed("") => vec![Line::from("")],
-            Cow::Borrowed(s) => s.lines().map(Line::from).collect(),
             Cow::Owned(s) if s.is_empty() => vec![Line::from("")],
-            Cow::Owned(s) => s.lines().map(|l| Line::from(l.to_owned())).collect(),
+            Cow::Borrowed(s) => s
+                .split('\n')
+                .map(|s| Line::from(s.strip_suffix('\r').unwrap_or(s)))
+                .collect(),
+            Cow::Owned(s) => s
+                .split('\n')
+                .map(|s| Line::from(s.strip_suffix('\r').unwrap_or(s).to_owned()))
+                .collect(),
```

Comment:
I've added a comment to explain. `lines()` eats trailing newlines, which means that if you later use `write!()` to append to the `Text`, your added text gets pushed onto the end of the previous line, instead of being added on the next line where it belongs.

> Is the handling for the empty string actually necessary?

I believe it is just a performance optimization, one I didn't change from the prior version of this code.